contracts/MarginBase.sol

```diff
@@ -412,4 +454,148 @@ contract MarginBase is MinimalProxyable {
+            IExchangeRates(
+                addressResolver.requireAndGetAddress(
+                    "ExchangeRates",
+                    "MarginBase: Could not get ExchangeRates"
+                )
+            );
+    }
+
+    /*///////////////////////////////////////////////////////////////
+                            Limit Orders
+    ///////////////////////////////////////////////////////////////*/
+
+    /// @notice limit order logic condition checker
+    /// @param _orderId: key for an active order
+    function validOrder(uint256 _orderId) public view returns (bool) {
+        Order memory order = orders[_orderId];
+
+        bytes32 currencyKey = futuresMarket(order.marketKey).baseAsset();
+        // Get exchange rate for 1 unit
+        uint256 price = exchangeRates().effectiveValue(
+            currencyKey,
+            1e18,
+            "sUSD"
+        );
+
+        if (order.sizeDelta > 0) {
+            // Long
+            return price <= order.desiredPrice;
+        } else if (order.sizeDelta < 0) {
+            // Short
+            return price >= order.desiredPrice;
+        } else {
+            revert("Order size 0");
+        }
+    }
+
+    /// @notice register a limit order internally and with gelato
+    /// @param _marketKey: synthetix futures market id/key
+    /// @param _marginDelta: amount of margin (in sUSD) to deposit or withdraw
+    /// @param _sizeDelta: denoted in market currency (i.e. ETH, BTC, etc), size of futures position
+    /// @param _limitPrice: expected limit order price
+    /// @return orderId contract interface
+    function placeOrder(
+        bytes32 _marketKey,
+        int256 _marginDelta,
+        int256 _sizeDelta,
+        uint256 _limitPrice
+    ) external onlyOwner returns (uint256) {
+        // if more margin is desired on the position we must commit the margin
+        if (_marginDelta > 0) {
+            // ensure margin doesn't exceed max
+            if (_abs(_marginDelta) > freeMargin())
+                revert InsufficientFreeMargin(freeMargin(), _abs(_marginDelta));
+            committedMargin += _abs(_marginDelta);
+        }
+
+        bytes32 taskId = IOps(ops).createTask(
+            address(this),
+            this.executeOrder.selector,
+            address(this),
+            abi.encodeWithSelector(this.checker.selector, orderId)
+        );
+
+        orders[orderId] = Order({
+            marketKey: _marketKey,
+            marginDelta: _marginDelta,
+            sizeDelta: _sizeDelta,
+            desiredPrice: _limitPrice,
+            gelatoTaskId: taskId
+        });
+
+        return orderId++;
+    }
+
+    /// @notice cancel a gelato queued order
+    /// @param _orderId: key for an active order
+    function cancelOrder(uint256 _orderId) external onlyOwner {
+        Order memory order = orders[_orderId];
+
+        // if margin was committed, free it
+        if (order.marginDelta > 0) {
+            committedMargin -= _abs(order.marginDelta);
```

Comment:
Maybe I am misunderstanding, but a negative `marginDelta` would indicate an order which will withdraw margin from a position. If that is the case, we would not need to alter the committed margin. 
-
Example: single order to withdraw `half the margin` in a position and reduce sizeDelta to 1x LONG:
In that case, no margin should be committed, right? If I did that, and then called this function, it would attempt to subtract `half the margin` in that position from `committedMargin` which is zero, resulting in a revert

Reply:
I do see above we check that marginDelta is > 0. So given that, do we need to call `_abs()` since it should not be negative?

Reply:
Yeah the `order.marginDelta > 0` makes sure this flow only happens if margin is being deposited. The reason for `_abs()` is because I'm implicitly using it to convert `int` to `uint` (knowing the value is already positive).